pkg/controller/pod_volume_backup_controller.go

```diff
@@ -18,330 +18,220 @@ package controller
-	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/vmware-tanzu/velero/internal/credentials"
 	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
-	velerov1client "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned/typed/velero/v1"
-	informers "github.com/vmware-tanzu/velero/pkg/generated/informers/externalversions/velero/v1"
 	listers "github.com/vmware-tanzu/velero/pkg/generated/listers/velero/v1"
 	"github.com/vmware-tanzu/velero/pkg/metrics"
 	"github.com/vmware-tanzu/velero/pkg/restic"
 	"github.com/vmware-tanzu/velero/pkg/util/filesystem"
 	"github.com/vmware-tanzu/velero/pkg/util/kube"
-
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type podVolumeBackupController struct {
-	*genericController
-
-	podVolumeBackupClient velerov1client.PodVolumeBackupsGetter
-	podVolumeBackupLister listers.PodVolumeBackupLister
-	podLister             corev1listers.PodLister
-	pvcLister             corev1listers.PersistentVolumeClaimLister
-	pvLister              corev1listers.PersistentVolumeLister
-	kbClient              client.Client
-	nodeName              string
-	metrics               *metrics.ServerMetrics
-	credentialsFileStore  credentials.FileStore
-
-	processBackupFunc func(*velerov1api.PodVolumeBackup) error
-	fileSystem        filesystem.Interface
-	clock             clock.Clock
-}
-
-// NewPodVolumeBackupController creates a new pod volume backup controller.
-func NewPodVolumeBackupController(
-	logger logrus.FieldLogger,
-	podVolumeBackupInformer informers.PodVolumeBackupInformer,
-	podVolumeBackupClient velerov1client.PodVolumeBackupsGetter,
-	podInformer cache.SharedIndexInformer,
-	pvcInformer corev1informers.PersistentVolumeClaimInformer,
-	pvInformer corev1informers.PersistentVolumeInformer,
-	metrics *metrics.ServerMetrics,
-	kbClient client.Client,
-	nodeName string,
-	credentialsFileStore credentials.FileStore,
-) Interface {
-	c := &podVolumeBackupController{
-		genericController:     newGenericController(PodVolumeBackup, logger),
-		podVolumeBackupClient: podVolumeBackupClient,
-		podVolumeBackupLister: podVolumeBackupInformer.Lister(),
-		podLister:             corev1listers.NewPodLister(podInformer.GetIndexer()),
-		pvcLister:             pvcInformer.Lister(),
-		pvLister:              pvInformer.Lister(),
-		kbClient:              kbClient,
-		nodeName:              nodeName,
-		metrics:               metrics,
-		credentialsFileStore:  credentialsFileStore,
-
-		fileSystem: filesystem.NewFileSystem(),
-		clock:      &clock.RealClock{},
-	}
-
-	c.syncHandler = c.processQueueItem
-	c.cacheSyncWaiters = append(
-		c.cacheSyncWaiters,
-		podVolumeBackupInformer.Informer().HasSynced,
-		podInformer.HasSynced,
-		pvcInformer.Informer().HasSynced,
-	)
-	c.processBackupFunc = c.processBackup
-
-	podVolumeBackupInformer.Informer().AddEventHandler(
-		cache.ResourceEventHandlerFuncs{
```

Comment:
In previous version, controller only care about Create and Update event of Pod Volume Backup. 
Can the new version controller still keep the event filter function?

Reply:
I'm afraid I don't understand your question, "Can the new version controller still keep the event filter function?"

Did you want me to keep the `podVolumeBackupInformer.Informer().AddEventHandler(...)` function?

Reply:
Yes. Event handler function filters out the Delete event.
But this may be a minor change. Even Delete event enters reconcile loop, it should not do anything harmful.